src/mines.js

```diff
@@ -0,0 +1,397 @@
+let win;
+let timerVar;
+let amountOfFlags;
+let amountOfMines;
+let seconds = 0;
+let minutes = 0;
+let end;
+let rows;
+let columns;
+let checkpoint = true;
+let bombed = false;
+
+function remove(arr, value) {
+  for (let i = 0; i < arr.length; i += 1) {
+    if (arr[i] === value) {
+      arr.splice(i, 1);
+    }
+  }
+}
+
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function checkIfHasMines(i1, i2) {
+  if (
+    document.querySelector(`[data-order = "${i1}_${i2}"]`) !== null &&
+    parseInt(
+      document
+        .querySelector(`[data-order = "${i1}_${i2}"]`)
+        .getAttribute("data-mined"),
+      2
+    ) === 1
+  ) {
+    return true;
+  }
+  return false;
+}
+function checkIfHasMinesObject(cell) {
+  return parseInt(cell.getAttribute("data-mined"), 2) === 1;
+}
+
+function revealCell(row, coll) {
+  document
+    .querySelector(`[data-order = '${row}_${coll}']`)
+    .classList.add("cell-opened");
+  document
+    .querySelector(`[data-order = '${row}_${coll}']`)
+    .classList.remove("cell-not_opened");
+  const mineCount = document
+    .querySelector(`[data-order = '${row}_${coll}']`)
+    .getAttribute("data-amount");
+  if (
+    mineCount > 0 &&
+    document.querySelector(`[data-order = '${row}_${coll}']`).firstChild ===
+      null
+  ) {
+    document
+      .querySelector(`[data-order = '${row}_${coll}']`)
+      .insertAdjacentHTML(
+        "beforeend",
+        `<span class="counter">${mineCount}</span>`
+      );
+  }
+}
+
+function createMines() {
+  for (let i = 0; i < amountOfMines; i += 1) {
+    const randomX = getRandomInt(0, rows - 1);
+    const randomY = getRandomInt(0, columns - 1);
+    const status = checkIfHasMines(randomX, randomY);
+    if (!status) {
+      document
+        .querySelector(`[data-order = "${randomX}_${randomY}"]`)
+        .setAttribute("data-mined", 1);
+      document
+        .querySelector(`[data-order = "${randomX}_${randomY}"]`)
+        .insertAdjacentHTML(
+          "beforeend",
+          "<img src = 'https://static.thenounproject.com/png/965385-200.png' class = 'bomb-pic'></img>"
+        );
+    } else {
+      i -= 1;
+      continue;
+    }
+  }
+  for (let cellRow = 0; cellRow < rows; cellRow += 1) {
+    for (let cellColl = 0; cellColl < columns; cellColl += 1) {
+      let MineCounts = 0;
```

Comment:
How's that different to `amountOfMines`?

Reply:
This is used to track mines nearby the cell